plugins/search-backend/src/service/router.ts

```diff
@@ -154,9 +160,13 @@ export async function createRouter(
         const resultSet = await engine?.query(query, { token });
 
         res.send(filterResultSet(toSearchResults(resultSet)));
-      } catch (err) {
+      } catch (error) {
+        if (error instanceof MissingIndexError) {
+          res.status(400).json(error.message);
```

Comment:
Now that I think about it, should this be a 400 or should this still actually be a 500 of some kind?

If I recall correctly, there's already a 400 error in place if you try and search using a `type` that doesn't correspond to a registered collator ([e.g.](https://demo.backstage.io/api/search/query?types[]=not-real)).

In this case, something would have had to have gone wrong with the collator on the server for the error to happen, so maybe a `500` is more appropriate.

Reply:
Yep just throwing an `InputError` leads to a 400

Reply:
Those can have a cause as well, and might pass along a traceback and stuff in dev mode